src/Symfony/Component/Mailer/Bridge/Mailjet/Tests/Transport/MailjetApiTransportTest.php

```diff
@@ -85,6 +89,185 @@ public function testPayloadFormat()
         $this->assertEquals('Qux', $replyTo['Name']);
     }
 
+    public function testDoSendApiSuccess()
+    {
+        $json = json_encode([
+            'Messages' => [
+                'foo' => 'bar',
+            ],
+        ]);
+
+        $responseHeaders = [
+            'x-mj-request-guid' => ['baz'],
+        ];
+
+        $response = new MockResponse($json, ['response_headers' => $responseHeaders]);
+
+        $client = new MockHttpClient($response);
+
+        $transport = new MailjetApiTransport(self::USER, self::PASSWORD, $client);
+        $method = new \ReflectionMethod(MailjetApiTransport::class, 'doSendApi');
```

Comment:
Thanks for the changes. I think we can go further and use the public API, can you check https://github.com/alamirault/symfony/pull/1 and apply it if it looks good for you? Thanks.

Reply:
Indeed, it's clearer !